src/types.jl

```diff
@@ -81,7 +81,21 @@ function FrameCode(scope, src::CodeInfo; generator=false, optimize=true)
         end
     end
     used = find_used(src)
-    return FrameCode(scope, src, methodtables, breakpoints, used, generator)
+    framecode = FrameCode(scope, src, methodtables, breakpoints, used, generator)
+    if scope isa Method && !is_generated(scope) # TODO: Relax this
+        method = scope
+        for bp in unresolved_breakpoints
+            method.file === bp.file || continue
```

Comment:
This is fragile (not just for Base but also wrt symlinks for stdlibs, etc) but I think you know that already. Other than fixing those cases I think we just have to :crossed_fingers: